### PR TITLE
additions to sanity checking wiki

### PR DESCRIPTION
**Pull request by eleanorjboyd**
Add extra sections to sanity checking helping to explain process to beginners.

### CODE REVIEW

File: Sanity-Check.md [hunk 1]
```diff
@@ -1,3 +1,75 @@
...
+    - Try some command from the command palette
+    - Open / click around in the terminal
+
+---
+---
+## Getting Started
+
+When the step of sanity testing comes in endgame, sections are assigned based on device type. The devices tested are as follows with the compatible testing machine:
+
+| OS to Test      | Possible Testing Machine for Given OS |
```
Comment (amunger) [inline Sanity-Check.md:33]: Windows ARM? only alternate is apple silicon AFAIK

Comment (amunger) [reply Sanity-Check.md:33]: and Linux. Is there a reason this doesn't just match the list of OS's to test below?

Comment (TylerLeonhardt) [reply Sanity-Check.md:33]: I'm a bit confused on the headings... it should read like this:
> If you have COLUMN 1, you can test COLUMN 2

Maybe 
```suggestion
| OS you are running | Additional builds you are able to test |
```

Comment (amunger) [reply Sanity-Check.md:33]: ooooh, yeah, I had it backwards

File: Sanity-Check.md [hunk 2]
```diff
@@ -1,3 +1,75 @@
+# Sanity Checking VS Code
+### Jump to a section:
+[About](#about) \
+[Getting Started](#getting-started) \
+[Steps to Test Per Build Type](#steps-to-test-per-build-type) 
+
+
+---
+---
```
Comment (TylerLeonhardt) [inline Sanity-Check.md:9]: is this needed to make the wiki work nicely?

File: Sanity-Check.md [hunk 3]
```diff
@@ -1,3 +1,75 @@
...
+## About
+
+#### What is sanity testing and why do we do it?
+Sanity testing is a process where we confirm our vscode installers work and that the final build can run. We sanity test to make sure the builds we ship work on the most basic level and there isn’t an immediate breaking change on VS Code. It also allows us to have human eyes on the actual build we ship and avoids the use of potentially flakey tests in the final stages of a release. (generally just make sure we don’t embarrass ourselves by shipping something that doesn’t even start).
```
Comment (TylerLeonhardt) [inline Sanity-Check.md:13]:
```suggestion

VS Code has an extensive release process, called [Endgame](https://github.com/Microsoft/vscode/wiki/Running-the-Endgame), that contains multiple testing processes to ensure that we deliver a product of the highest quality to our users. The final process a build must pass before being released is what we call "Sanity testing".

> NOTE: This testing is done for each platform build we ship.

Sanity testing is human run and is the opportunity to have a final set of eyes on a build before it is released to users. This testing helps catch any final issues that might have slipped through the rest of the testing processes and ensures that the installer install VS Code correctly and it runs as expect.
```

File: Sanity-Check.md [hunk 4]
```diff
@@ -1,3 +1,75 @@
...
+
+We are looking for errors, failures, or anything else that is not desired behavior. Examples are things not loading or errors when trying to open the build in the first place.
+
+#### What does it mean to sanity test and what are some things to try during sanity testing?
+
+- On the most basic level:  install, launch, then open a file or the about dialog
+- Other options to test a bit more:
+    - Click through all the items on the sidebar
+    - Try some command from the command palette
+    - Open / click around in the terminal
```
Comment (TylerLeonhardt) [inline Sanity-Check.md:25]:
```suggestion
    - Open / click around in the terminal
    - Install and use an extension
```

File: Sanity-Check.md [hunk 5]
```diff
@@ -1,3 +1,75 @@
...
+#### Assignment & Beginning
+From here, different OS sections are self-assigned and checked off once complete on the given release github issue. All OSs are tested for Archive, Server, and CLI. The OS tested are as follows:
+- Windows x32
+- Windows x64
+- Windows ARM64
+- Mac Intel
+- Mac ARM64
+- Linux x64
+- Linux Server
+- Linux CLI
```
Comment (TylerLeonhardt) [inline Sanity-Check.md:56]: These aren't really OSs. I would maybe do:
* Windows (64bit, 32bit & ARM)
* macOS (Intel & Silicon)
* Linux (various architectures)

File: Sanity-Check.md [hunk 6]
```diff
@@ -1,3 +1,75 @@
...
+
+
+
+Other system types have specific builds, these will be outlined in the release issue. Here are the types of builds and what this means:
+
+- Archive: The regular build (also referred to as client below)
+- Server: Running vscode on a server on a remote client.
+- CLI: Running vscode from the command line interface.
+- Universal Archive (for mac): Same as the regular archive but just a build that should work universally for all Macs.
```
Comment (TylerLeonhardt) [inline Sanity-Check.md:65]:
```suggestion
- Universal Archive (for macOS): Same as the regular archive but just a build that should work universally for both Intel and Apple Silicon based chips.
```

File: Sanity-Check.md [hunk 7]
```diff
@@ -1,3 +1,75 @@
...
+
+---
+---
+## Getting Started
+
+When the step of sanity testing comes in endgame, sections are assigned based on device type. The devices tested are as follows with the compatible testing machine:
+
+| OS to Test      | Possible Testing Machine for Given OS |
+| ----------- | ----------- |
+| Mac x64      | Windows x64, Windows x86, Linux x64, Linux Server, Linux CLI       |
```
Comment (roblourens) [inline Sanity-Check.md:35]: I feel like this table is adding a lot of noise. The takeaway is really that you can pretty much test anything from anything. Probably the only combos that don't work is testing a mac build when you only have linux, or going between architectures (sometimes). This also isn't a sanity-check-specific concept. In my mind the list of tools for cross-OS testing is more helpful

File: Sanity-Check.md [hunk 8]
```diff
@@ -1,3 +1,75 @@
...
+| Mac ARM   | Windows ARM, Linux Server, Linux CLI,        |
+| Windows x64 | Windows x86, Linux x64, Linux Server, Linux CLI |
+| Any Web Browser |  Windows x64, Windows x86, Linux x64, Linux Server, Linux CLI |
+
+credit to @tylerleonhardt for this table and the below tool descriptions.
+
+#### Tools for Testing Different OS
+* On macOS you can get a Parallels license to spin up VMs , you can get a Parallels license.
```
Comment (roblourens) [inline Sanity-Check.md:43]: duplicated "you can get a Parallels license"

File: Sanity-Check.md [hunk 9]
```diff
@@ -1,3 +1,75 @@
...
+| Mac x64      | Windows x64, Windows x86, Linux x64, Linux Server, Linux CLI       |
+| Mac ARM   | Windows ARM, Linux Server, Linux CLI,        |
+| Windows x64 | Windows x86, Linux x64, Linux Server, Linux CLI |
+| Any Web Browser |  Windows x64, Windows x86, Linux x64, Linux Server, Linux CLI |
+
+credit to @tylerleonhardt for this table and the below tool descriptions.
+
+#### Tools for Testing Different OS
+* On macOS you can get a Parallels license to spin up VMs , you can get a Parallels license. 
+* On Windows (virtualized or not), you can use the Sanity Testing WSL image created by @sbatten
```
Comment (roblourens) [inline Sanity-Check.md:44]: Link to this?

File: Sanity-Check.md (hunk 8 above)
Comment (roblourens) [reply Sanity-Check.md:43]: Mention that you can get this from our admin

File: Sanity-Check.md (hunk 5 above)
Comment (roblourens) [reply Sanity-Check.md:56]: I don't think we even need to list this out here

File: Sanity-Check.md [hunk 10]
```diff
@@ -1,3 +1,75 @@
...
+- Mac ARM64
+- Linux x64
+- Linux Server
+- Linux CLI
+
+
+
+Other system types have specific builds, these will be outlined in the release issue. Here are the types of builds and what this means:
+
+- Archive: The regular build (also referred to as client below)
```
Comment (roblourens) [inline Sanity-Check.md:62]: An archive is a build that doesn't have an installer (but I also am not sure this section adds much value)

File: Sanity-Check.md [hunk 11]
```diff
@@ -1,3 +1,75 @@
...
+## Getting Started
+
+When the step of sanity testing comes in endgame, sections are assigned based on device type. The devices tested are as follows with the compatible testing machine:
+
+| OS to Test      | Possible Testing Machine for Given OS |
+| ----------- | ----------- |
+| Mac x64      | Windows x64, Windows x86, Linux x64, Linux Server, Linux CLI       |
+| Mac ARM   | Windows ARM, Linux Server, Linux CLI,        |
+| Windows x64 | Windows x86, Linux x64, Linux Server, Linux CLI |
+| Any Web Browser |  Windows x64, Windows x86, Linux x64, Linux Server, Linux CLI |
```
[the remark this answers is not in the document]
Comment (TylerLeonhardt) [reply Sanity-Check.md:38]: devbox.microsoft.com

File: Sanity-Check.md (hunk 5 above)
Comment (TylerLeonhardt) [reply Sanity-Check.md:56]: It should be captured somewhere though... I keep having to remind people that they have more testing options than they think they do.

Comment (roblourens) [reply Sanity-Check.md:56]: The list of testing tools, sure, but this is just a list of the flavors we ship which is already in the endgame issue

**Pull request merged**